pkgs/applications/misc/bemenu/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchFromGitHub, cairo, cmake, libxkbcommon
+, pango, fribidi, harfbuzz, pcre, pkgconfig
+, ncursesSupport ? true, ncurses ? null
+, waylandSupport ? true, wayland ? null
+, x11Support ? true, xlibs ? null, xorg ? null
+}:
+
+assert ncursesSupport -> ncurses != null;
+assert waylandSupport -> wayland != null;
+assert x11Support -> xlibs != null && xorg != null;
+
+stdenv.mkDerivation rec {
+  pname = "bemenu";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Cloudef";
+    repo = "bemenu";
+    rev = "33e540a2b04ce78f5c7ab4a60b899c67f586cc32";
+    sha256 = "11h55m9dx6ai12pqij52ydjm36dvrcc856pa834njihrp626pl4w";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig pcre ];
+
+  buildInputs = with stdenv.lib; [
+    cairo
+    fribidi
+    harfbuzz
+    libxkbcommon
+    pango
+  ] ++ optionals ncursesSupport [ ncurses ]
+    ++ optionals waylandSupport [ wayland ]
+    ++ optionals x11Support [
+      xlibs.libX11 xlibs.libXinerama xlibs.libXft
+      xorg.libXdmcp xorg.libpthreadstubs xorg.libxcb
```

Comment:
Seems to build fine without the xorg dependencies for me though someone else should probably check this as I'm not actually running X (it does claim to be building the x11 backend though)

Reply:
@alexarice Sure, this makes sense. Removed the null from dependencies.

Reply:
Does your build break without the `xorg` dependencies, or are they needed at run time or something similar

Reply:
Setting `x11Support` to false correctly builds the binary without X11 support.

Reply:
Well, it really breaks eval when removing those null checks. I re-added them, I think the tests should pass now.

Reply:
I'm still not being clear, I can remove the dependencies starting with xorg from the nix file and build it with x11 support